File: frontend/src/api/dto/notificationTeamPreferenceUpdate.ts
/**
 * Mute or unmute one of the current user's teams
 */
export interface NotificationTeamPreferenceUpdate {
  /**
   * Team slug — a team the user belongs to
   * @pattern \S
   */
  teamSlug: string
  /** Whether to mute its announcements */
  muted: boolean
}

File: backend/src/main/java/fr/pedalons/dto/notifications/request/NotificationTeamPreferenceUpdate.java
package fr.pedalons.dto.notifications.request;

import fr.pedalons.dto.validation.ValidateSchema;
import jakarta.validation.constraints.NotBlank;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Mute or unmute one of the current user's teams")
@ValidateSchema
public record NotificationTeamPreferenceUpdate(
    @Schema(description = "Team slug — a team the user belongs to", required = true) @NotBlank
        String teamSlug,
    @Schema(description = "Whether to mute its announcements", required = true) boolean muted) {}
